pilot/pkg/xds/bench_test.go

```diff
@@ -185,55 +142,97 @@ func buildTestEnv(t test.Failer, cfg []model.Config, input ConfigInput) model.En
 	return env
 }
 
+type FakeXdsUpdater struct {
+}
+
+func (fx *FakeXdsUpdater) EDSUpdate(_, _ string, _ string, _ []*model.IstioEndpoint) error {
+	return nil
+}
+
+func (fx *FakeXdsUpdater) ConfigUpdate(_ *model.PushRequest) {
+}
+
+func (fx *FakeXdsUpdater) ProxyUpdate(_, _ string) {
+}
+
+func (fx *FakeXdsUpdater) SvcUpdate(_, _ string, _ string, _ model.Event) {
+}
+
 // ConfigInput defines inputs passed to the test config templates
 // This allows tests to do things like create a virtual service for each service, for example
 type ConfigInput struct {
-	// A list of hostname of allservices
-	Services []string
+	// Name of the test
+	Name string
+	// Name of the test config file to use. If not set, <Name> is used
+	ConfigName string
+	// Number of services to make
+	Services int
+	// Type of proxy to generate configs for
+	ProxyType model.NodeType
 }
 
-// Setup test builds a mock test environment.
-// TODO make this configurable with Istio config, different service setups, etc
-func setupTest(t test.Failer, testName string) (model.Environment, core.ConfigGenerator, model.Proxy) {
-	proxy := model.Proxy{
-		Type:        model.SidecarProxy,
+// Setup test builds a mock test environment. Note: push context is not initialized, to be able to benchmark separately
+// most should just call setupAndInitializeTest
+func setupTest(t testing.TB, config ConfigInput) (*model.Environment, core.ConfigGenerator, *model.Proxy) {
+	proxyType := config.ProxyType
+	if proxyType == "" {
+		proxyType = model.SidecarProxy
+	}
+	proxy := &model.Proxy{
+		Type:        proxyType,
 		IPAddresses: []string{"1.1.1.1"},
 		ID:          "v0.default",
 		DNSDomain:   "default.example.org",
 		Metadata: &model.NodeMetadata{
-			Namespace: "not-default",
+			Namespace: "default",
```

Comment:
I believe the `IstioVersion` has to be set here as well, or the one below has to be taken into account in https://github.com/istio/istio/blob/9d7cf2d63af42de63ec807e95224066b03fe55e8/pilot/pkg/model/envoyfilter.go#L123; otherwise listener patches and probably other codepaths won't run.